### PR TITLE
Refactor model to make things cleaner

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentPlugin.java [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
 
 /**
  * Results in adding a task with name {@value org.jboss.gm.analyzer.alignment.AlignmentTask#NAME}.
- * It also creates the basic alignment.json file that {@value org.jboss.gm.analyzer.alignment.AlignmentTask#NAME} of each project adds to
+ * It also creates the basic alignment.json file that {@value org.jboss.gm.analyzer.alignment.AlignmentTask#NAME} of each
+ * project adds to
  *
  */
-public class AlignmentPlugin implements Plugin<Project> {
+public class AlignmentPlugin implements Plugin<org.gradle.api.Project> {
```
Comment (geoand) [inline analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentPlugin.java]: Can we get rid of the package name here an in some of the other places it appears?

File: common/src/main/java/org/jboss/gm/common/alignment/Module.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Map;
+
+import org.commonjava.maven.atlas.ident.ref.ProjectVersionRef;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * @author <a href="claprun@redhat.com">Christophe Laprun</a>
+ */
+public class Module {
```
Comment (geoand) [inline common/src/main/java/org/jboss/gm/common/alignment/Module.java]: I take it this was broken out in order to allow `AlignedProject` to extend from it, correct?

**Comment by geoand** [comment]
I don't have strong opinions on the naming matter, whatever others decide I'm good with :)

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentPlugin.java (hunk 1 above)
Comment (metacosm) [reply analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentPlugin.java]: Yes, that's what I'm actually fixing right now :)

File: common/src/main/java/org/jboss/gm/common/alignment/Module.java (hunk 2 above)
Comment (metacosm) [reply common/src/main/java/org/jboss/gm/common/alignment/Module.java]: Yes.

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentPlugin.java (hunk 1 above)
Comment (geoand) [reply analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentPlugin.java]: :+1:

File: common/src/main/java/org/jboss/gm/common/alignment/Module.java (hunk 2 above)
Comment (geoand) [reply common/src/main/java/org/jboss/gm/common/alignment/Module.java]: OK

**Comment by rnc** [comment]
What are "AlignedProject" and "Module" meant to represent? Is AlignedProject the result of aligned a (sub)project? What about Module?

**Comment by metacosm** [comment]
@rnc I've started working on javadoc locally but I wanted to get these changes in, first, if they were agreeable. `AlignedProject` represents the root project / maven build being aligned, whereas `Module` represents sub-projects / children modules. If you can think of better names, let's change them!

**Comment by metacosm** [comment]
retest this please

**Pull request merged**